docs/source/transaction_family_specifications/sawtooth_burrow_evm_family.rst

```diff
@@ -0,0 +1,269 @@
+  or account balances, transferring value between accounts is not meaningful.
+
+Execution
+=========
+
+Transaction execution shall follow a simplified version of the Ethereum model
+described below:
+
+1. The sender address shall be calculated by taking the rightmost 160 bits of
+   the SHA3 hash of the public key used to submit the transaction. This public
+   key is included in the transaction header.
+2. If the `to` field is set, the transaction is interpreted as a “message call”
+   transaction. If not, it is a “contract creation” transaction.
+3. If the transaction is a “message call” transaction,
+
+    a. The receiver’s account will be retrieved from state using the address in
+       the `to` field. If the address is invalid, the account does not exist, or
+       the account does not contain any code, the transaction is invalid.
+    b. The sender’s account will be retrieved from state using the sender
+       address. If the account does not exist, the transaction is invalid.
+    c. The EVM will be called using:
+
+        - The sender account for the `caller` argument
+        - The receiver account for the `callee` argument
+        - The `code` field in the receiver’s account for the `code` argument
+        - The `data` field in the transaction payload for the `input` argument.
+          If no data field is set, the transaction is invalid.
+        - 0 for the `value` argument.
+        - The `gas_limit` field in the transaction payload for the `gas`
+          argument.
+
+    d. The resulting output from the EVM shall be discarded because a mechanism
+       for returning data to the client does not exist yet.
+    e. The sender and receiver accounts and associated storage shall be pushed
+       to state.
+
+4. If the transaction is a “contract creation” transaction,
+
+    a. If the sender address does not exist yet, an Externally Owned Account
+       shall be created at the address.
+    b. If the sender address does exist, a Contract Account shall be created at
+       a new address derived from the sender account as described above. The
+       sender account’s nonce shall then be incremented.
+    c. If the `init` field is set in the transaction payload, the EVM will be
+       called using:
+
+        - The sender account for the `caller` argument
+        - The sender account for the `callee` argument
+        - The `init` field for the `code` argument
+        - `nil` for the `input` argument.
+        - 0 for the `value` argument.
+        - The `gas_limit` field in the transaction payload for the `gas`
+          argument.
+
+    d. The resulting output from the EVM shall be stored in the `code` field of
+       the newly created account.
+    e. The sender account and the newly created account shall be pushed to
+       state.
+
+5. If an error occurs while the EVM is executing, the transaction is invalid.
+6. If a transaction is invalid, no updates to state will be made.
```

Comment:
Hmm, I see what you mean. I wrote this with the ethereum model in mind. In the ethereum model, changes to state can be made even if a transaction fails to commit, for example if an out of gas exception occurs. I wanted to clarify that this is not he case here. Although, in that case, we may not want to consider a transaction invalid in that case within the context of sawtooth... I need to think about that some more...